source/common/res/features/budget-daily-spending/main.js

```diff
@@ -21,11 +21,24 @@
         },
 
         observe(changedNodes) {
-          if (changedNodes.has('budget-inspector')) {
+          if (changedNodes.has('navlink-budget active') ||
+              changedNodes.has('budget-table-cell-available-div user-data') ||
+              changedNodes.has('budget-inspector')) {
             ynabToolKit.dailySpending.invoke();
           }
         },
 
+        addBudgetVersionIdObserver() {
+          let applicationController = ynabToolKit.shared.containerLookup('controller:application');
+          applicationController.addObserver('budgetVersionId', function () {
+            Ember.run.scheduleOnce('afterRender', this, resetBudgetViewCheckCreditBalances);
+          });
```

Comment:
This function needs to be in your feature code. That way your feature can do whatever it needs to do when the budget ID changes. In the features I've worked on I  name them resetBudgetViewxxx where xxx is the name of the feature. For this feature it would be resetBudgetViewDailySpending.